test/unit_test.ts

```diff
@@ -3,245 +3,178 @@ import chai = require('chai');
+          b: string;
       }
-    `,
-        ['A', 'A.constructor(prop:number, arg1:number)', 'A.prop:number']);
-  });
-
-  it('should include protected properties defined via constructor', () => {
-    check(
-        `
-      export class A {
-        constructor(protected prop: number, arg1: number) {}
-      }
-    `,
-        ['A', 'A.constructor(prop:number, arg1:number)', 'A.prop:number //protected']);
-  });
-
-  it('should include default value of public properties defined via constructor', () => {
-    check(
-        `
-      export class A {
-        constructor(public prop = 3) {}
-      }
-    `,
-        ['A', 'A.constructor(prop:any=3)', 'A.prop:any=3']);
-  });
-
-  it('should ignore private properties defined via constructor', () => {
-    check(
-        `
-    export class A {
-      constructor(private prop = 3) {}
-    }
-  `,
-        ['A', 'A.constructor(prop:any=3)']);
-  });
-
-  it('should recognize optional parameters of functions, methods and constructors', () => {
-    check(
-        `
-    export class A {
-      constructor(arg?: number) {}
-      methodA(arg?:number) {}
-    }
-
-    export function b(arg?: string) {}
-  `,
-        ['A', 'A.constructor(arg?:number)', 'A.methodA(arg?:number):any', 'b(arg?:string):any']);
+    `;
+    check({'file.d.ts': input}, expected);
   });
 });
 
-function check(contents: string, expected: string[]) {
-  var mockHost: any = {
+function getMockHost(files: {[name: string]: string}): ts.CompilerHost {
+  return {
     getSourceFile: (sourceName, languageVersion) => {
-      if (sourceName !== 'file.ts') return undefined;
-      return ts.createSourceFile(sourceName, contents, languageVersion, true);
+      if (!files[sourceName]) return undefined;
+      return ts.createSourceFile(sourceName, files[sourceName], languageVersion, true);
     },
-    writeFile(name, text, writeByteOrderMark) {},
-    fileExists: (filename) => filename === 'file.ts',
-    readFile: (filename) => contents,
+    writeFile: (name, text, writeByteOrderMark) => {},
+    fileExists: (filename) => !!files[filename],
+    readFile: (filename) => normalizeText(files[filename]),
     getDefaultLibFileName: () => 'lib.ts',
     useCaseSensitiveFileNames: () => true,
     getCanonicalFileName: (filename) => filename,
     getCurrentDirectory: () => './',
     getNewLine: () => '\n',
   };
-  const actual = main.publicApiInternal(mockHost, 'file.ts');
-  chai.assert.deepEqual(actual, expected);
-}
\ No newline at end of file
+}
+
+function check(files: {[name: string]: string}, expected: string) {
+  const actual = main.publicApiInternal(getMockHost(files), 'file.d.ts', {});
+  chai.assert.equal(normalizeText(actual), normalizeText(expected));
+}
+
+function checkThrows(files: {[name: string]: string}, error: string) {
+  chai.assert.throws(
+      () => { console.log(main.publicApiInternal(getMockHost(files), 'file.d.ts', {})); }, error);
+}
+
+function normalizeText(text: string) {
```

Comment:
more descriptive name would be `stripExtraIndentation` or something similar, or at least add a comment block above explaining what this does function does because "normalizeText" is too generic.